data_management/opensearch_indexer/opensearch_indexer/text_extraction.py

```diff
@@ -76,5 +80,56 @@ def extract_text(file_stream: bytes, file_extension: str) -> str:
     ) as temp:
         temp.write(file_stream)
         temp.flush()
-        context = textract.process(temp.name)
-    return context.decode("utf-8")
+        file_path = temp.name
+
+        try:
+            context = textract.process(file_path)
+            return context.decode("utf-8")
+
+        except Exception as e:
+            logger.warning(f"Textract failed on {file_path}: {e}")
+
+            if file_extension not in CONVERT_MAP:
+                raise e
+            try:
+                logger.info("Attempting to convert via LibreOffice...")
+                converted_path = convert_failed_files(file_path)
+                logger.info(f"Converted to: {converted_path}")
+                text = textract.process(converted_path)
+                return text.decode("utf-8")
+            except Exception as convert_err:
+                logger.error(f"LibreOffice conversion failed: {convert_err}")
```

Comment:
this log error is not necessarily true, it could be that conversion faield after converting but text extraction still failed. Can move the try except around the `convert_failed_files` line specifically